doc/gui/extension/example_library/front-end/src/BasicTable.tsx

```diff
@@ -0,0 +1,133 @@
+import React, { useEffect, useMemo, useState } from "react";
+import {
+    createRequestDataUpdateAction,
+    RowType,
+    RowValue,
+    useDispatch,
+    useDispatchRequestUpdateOnFirstRender,
+    useModule,
+} from "taipy-gui";
+
+interface BasicTableProps {
+    id?: string;
+    updateVarName?: string;
+    updateVars?: string;
+    data: Record<string, Record<string, unknown>>;
+    rowsPerPage?: number;
+}
+
+const BasicTable = (props: BasicTableProps) => {
+    const {
+        data,
+        rowsPerPage,
+        updateVarName = "",
+        updateVars = "",
+        id
+    } = props;
+    const [value, setValue] = useState<Record<string, unknown>>({});
+    const [currentPage, setCurrentPage] = useState(1);
+    const dispatch = useDispatch();
+    const module = useModule();
+    const refresh = data?.__taipy_refresh !== undefined;
+    useDispatchRequestUpdateOnFirstRender(dispatch, id, module, updateVars, updateVarName);
```

Comment:
is `updateVarName` useful here or does it generate an unnecessary (and not valid) request ?

Reply:
I've removed it